### DISCUSSION TITLE
Patreon Funds - what to use for

### DISCUSSION THREAD

**Discussion by ian-jorgensen** [SynthstromAudible/DelugeFirmware#205] [General]
Hey everyone.

So, it looks like the Patreon currently has 141 patrons (and growing), contributing around $400 a month at the moment, with about $800 currently sitting in it.

I'll share specifics and screenshot the dashboard details when making payouts, so peeps can see :)

Would love to hear ideas about how best to use this. We could work out a way of dividing these amounts amongst those contributing, or putting larger amounts to things which perhaps make everyones lives easier?

One idea suggested was using some of this money to sorting out debugging hardware mods/cables and shipping to us for those who are actively contributing. 

We could also use the money to pay for access to extra features (actions) on the GitHub, or enhanced stuff on Discord? 

If we were to split the money between devs, I think perhaps the easiest way is to perhaps have a monthly poll? or something, where everyone votes for who were the top 10 contributors that month, we then just split the money evenly between those ten. This saves having to try to work out the individual value of everyones work, and just sorta gives a little bonus to the most active contributors that month (I assume this will change regularly) - currently around $40 each, enough money for a meal, nice bottle of wine/spirits, dozen beers, box of chocolates, or the most boujee orange juice/bottled water on earth. 

I think the community would love to see cash going directly into the hands of coders, even though not a huge amount of money...I think people would love to see there be a direct thank you between the community and devs. Would be nice to even share photos of devs sharing what they spend their payments on the Patreon.  I think the community picture it as 'buying a drink' for the devs, so haha, sharing literal photos of that I feel could really reflect that love :)

Obvs, it sucks to make those who don't make the 'top ten' each month to feel they're not valued, and it bums me out thinking people might think that way when of course that isn't the case...but yeah, I'm not sure of a better way?

**Comment by weavermedia** [comment]
Paying for service costs like GitHub and something to help those who want debugging setups sounds like a great idea.

A free supply of 3D-printed side cheeks for debugging ports would be handy! 😀

It would be cool to let the patrons and the rest of the Deluge community know about the amazing people working on the project.

People who made strides to get the build process working on Mac. People who set up the amazing cheap alternative for debugging. People who spent time wrangling Rohan's one-man code into an organized, workable open source codebase. People who are modularizing and streamlining the code to make way for faster, more efficient updates in the future.

Adding new features too, of course. We all love shiny new features and so many incredible things have been added already. But that's only part of the story. There's so much more going on that the regular users just don't know about.

**Reply by ian-jorgensen** [reply]
A supply of side cheeks is a good idea.

Maybe in like 3-4 months time or something, we could start a 'profiles' thing where we share a photo/bio of devs (who are happy to be involved) and do like a little IV where they talk about their contributions to the project etc as well as what they do outside of Synthstrom - a different person each month. 

Would be really nice to, as you say, make sure that the community understands all the 'non-glamour' work going down :)

I did have this vision of a booth of Superbooth next year just being helmed by community dev's, haha, I can just sit around eating bratwursts and drinking Fritz Kolas in the background while a bunch of dev's talk about open-source :) It's possible we could set aside some money for that? perhaps a contribution to any coder who wanted to go? (just an idea - again, I appreciate many people involved in this project couldn't possibly go even with a subsidy)

**Comment by PaulFreund** [comment]
Unfortunately it is very very hard to assess software work fairly. The amount of work that had to be put in is individual and can be very hard to grasp. Same with the positive impact on the application, there is just no simple metric. In addition the precieved value for end user features is of course a lot higher than the back end that enabled them.

While being a nice idea for the individuals I think voting might lead to people ending up hurt for the wrong reasons. Maybe if we want to do individual payouts we should do it equally for all contributors and make that clear beforehand so nobody is disappointed.

I personally don't need individual payouts but I understand that everyone has a different motivation. Apart from infrastructure stuff we might need I like the idea of saving up for paying for developers SWD upgrades, Debuggers, replacement side panels and stuff like repairs for units that were damaged by accident.

**Reply by ian-jorgensen** [reply]
Yeah, though I love the idea of peeps getting bought drinks by the community, it is fraught with issues, a utopian idea that will likely end in wrongly appearing to celebrate some and not others when this is a huge collective effort.

I think you're right in that we use it as a fund for covering associated costs and making this whole thing work better rather than using for small rewards. 

Of course, haha, determining how much work someone needs to put in before getting their debugging funded etc, is a whole other can of worms ;)

**Comment by sichtbeton** [comment]
All of the above assessments are true. Paying for general costs that benefit the whole community like GitHub server time or other services would be the best way to go atm. 

What other community costs could be covered? Supplying debugging kits is a wonderful idea. 

As for deepen the community spirit: a photo wall or something to actually see the devs / testers / other contributors (if they choose to do so) is something that gives a face to a faceless GitHub project.

**Comment by m-m-adams** [comment]
Yeah I agree that directly paying devs would probably end up distorting efforts in the project. A contributor vote would be the only 'fair' way to do it but still has some risk of becoming toxic. Even things like feature bounties get problematic with so many people volunteering - if for example Alice 'completes' a bounty but it relies on Bob's plugin framework which relies on Charlie's allocator and everyone used David's refactor then who gets paid?



I really like the idea of distributing side cheeks with a debug cutout though!

**Comment by seangoodvibes** [comment]
I agree about the unfairness about distributing funds.

I think reinvesting those funds in the project makes the most sense to me.

I like the suggestion of distributing debug cheeks! It will help with development so much.

**Comment by soymonitus** [comment]
I love the idea to make developers' lifes easier so having debugging kits for contributors sounds like a great idea and also for Github Actions or anything that makes contributing easier

**Comment by ian-jorgensen** [comment]
Alrighty. Seems this idea of funding debugging kits seems to be the way.
Does someone have a good idea of how I can determine who should be able to receive those?

**Reply by weavermedia** [reply]
One qualification could be a merged PR.

Also, speaking for myself, I'm fine to buy all the parts - I just don't want to cut my case. So I'd only need a printed side panel.
